modules/apps/portal-cache/portal-cache-ehcache-impl/src/main/java/com/liferay/portal/cache/ehcache/internal/BaseEhcachePortalCacheManager.java

```diff
@@ -452,12 +470,27 @@ private void _reconfigEhcache(Configuration configuration) {
 					PortalCache<K, V> portalCache = fetchPortalCache(
 						portalCacheName);
 
+					boolean companyIdShardedCache = false;
+
 					if (portalCache != null) {
 						BaseEhcachePortalCache<K, V> baseEhcachePortalCache =
 							EhcacheUnwrapUtil.getWrappedPortalCache(
 								portalCache);
 
-						if (baseEhcachePortalCache != null) {
+						if ((baseEhcachePortalCache != null) &&
+							baseEhcachePortalCache.isSharded()) {
+
+							long companyId = _getShardedCacheCompanyId(
+								portalCacheName);
+
+							if (companyId > 0) {
+								((ShardedEhcachePortalCache)
+									baseEhcachePortalCache).removeEhcache(
+										companyId);
+								companyIdShardedCache = true;
+							}
+						}
```

Comment:
@marianoalvarosaiz, please do SF for the part. It should have one emptyLine before `companyIdShardedCache = true;`.

In addition, for the code `((ShardedEhcachePortalCache)baseEhcachePortalCache)`, please firstly do the conversion, and then invoke `removeEhcache`, just like:
```
ShardedEhcachePortalCache shardedEhcachePortalCache = (ShardedEhcachePortalCache)baseEhcachePortalCache;

shardedEhcachePortalCache.removeEhcache()
```